File: btpy_lib/btpy/src/btpy/btpy_list/mod/count/count.py



def count(STRUCTURE:list|dict, CHECKER)\
        -> int:
    """
    Funcion que cuenta el numero de 
    verificaciones con una funcion 
    checker enviada.
    """
    n = 0
    array = []
    if(type(STRUCTURE) == dict):
        array = STRUCTURE.values()
    else:
        array = STRUCTURE
    for e in array:
        if(CHECKER(e)):
            n += 1
    return n

File: btjs_page/btjs/mod/structures/mod/count.js

import { is_object_js } 
    from "../../checkers/mod/is_object_js.js"

/*
Funcion que cuenta el numero de 
verificaciones con una funcion 
checker enviada en un array o jsobject.
    Envia un lambda con esta estructura:
        (e)=>{return e == condicion};
*/
export function count(STRUCTURE, 
        CHECKER) {
    let n = 0
    let array = []
    if(is_object_js(STRUCTURE)) {
        array = STRUCTURE.values() 
    } else {
        array = STRUCTURE
    }
    for(let i in array) {
        console.log("e: ", array[i])
        if(CHECKER(array[i])) {
            n += 1
        }
    }
    return n
}